modules/drivers/gnss/parser/newtonm2_parser.cc

```diff
@@ -0,0 +1,916 @@
+    novatel::ImuParameter param = novatel::GetImuParameter(imu_type);
+    AINFO << "IMU type: " << config::ImuType_Name(imu_type) << "; "
+          << "Gyro scale: " << param.gyro_scale << "; "
+          << "Accel scale: " << param.accel_scale << "; "
+          << "Sampling rate: " << param.sampling_rate_hz << ".";
+
+    if (newtonm2::is_zero(param.sampling_rate_hz)) {
+      AERROR_EVERY(5) << "Unsupported IMU type: "
+                      << config::ImuType_Name(imu_type);
+      return false;
+    }
+    gyro_scale_ = param.gyro_scale * param.sampling_rate_hz;
+    accel_scale_ = param.accel_scale * param.sampling_rate_hz;
+    imu_measurement_hz_ = param.sampling_rate_hz;
+    imu_measurement_span_ = 1.0 / param.sampling_rate_hz;
+    imu_.set_measurement_span(imu_measurement_span_);
+  }
+
+  double time = imu->gps_week * newtonm2::SECONDS_PER_WEEK + imu->gps_seconds;
+  if (imu_measurement_time_previous_ > 0.0 &&
+      fabs(time - imu_measurement_time_previous_ - imu_measurement_span_) >
+          1e-4) {
+    AWARN_EVERY(5) << "Unexpected delay between two IMU measurements at: "
+                   << time - imu_measurement_time_previous_;
+  }
+  imu_.set_measurement_time(time);
+  switch (imu_frame_mapping_) {
+    case 5:  // Default mapping.
+      newtonm2::rfu_to_flu(imu->x_velocity_change * accel_scale_,
+                           -imu->y_velocity_change_neg * accel_scale_,
+                           imu->z_velocity_change * accel_scale_,
+                           imu_.mutable_linear_acceleration());
+      newtonm2::rfu_to_flu(imu->x_angle_change * gyro_scale_,
+                           -imu->y_angle_change_neg * gyro_scale_,
+                           imu->z_angle_change * gyro_scale_,
+                           imu_.mutable_angular_velocity());
+      break;
+    case 6:
+      newtonm2::rfu_to_flu(-imu->y_velocity_change_neg * accel_scale_,
+                           imu->x_velocity_change * accel_scale_,
+                           -imu->z_velocity_change * accel_scale_,
+                           imu_.mutable_linear_acceleration());
+      newtonm2::rfu_to_flu(-imu->y_angle_change_neg * gyro_scale_,
+                           imu->x_angle_change * gyro_scale_,
+                           -imu->z_angle_change * gyro_scale_,
+                           imu_.mutable_angular_velocity());
+      break;
+    default:
+      AERROR_EVERY(5) << "Unsupported IMU frame mapping: "
+                      << imu_frame_mapping_;
+  }
+  imu_measurement_time_previous_ = time;
+  return true;
+}
+
+bool NewtonM2Parser::HandleRawImu(const novatel::RawImu* imu) {
+  double gyro_scale = 0.0;
+  double accel_scale = 0.0;
+  float imu_measurement_span = 1.0 / 200.0;
+
+  if (newtonm2::is_zero(gyro_scale_)) {
+    novatel::ImuParameter param = novatel::GetImuParameter(imu_type_);
+
+    if (newtonm2::is_zero(param.sampling_rate_hz)) {
+      AERROR_EVERY(5) << "Unsupported IMU type ADUS16488.";
```

Comment:
wrong imu type here. why same error still keep from different PRs? please use ONE PR for your 
 all changes about same topic.

Reply:
I have Already modified it, please review again.